schema_based_on_diagram.sql

```diff
@@ -0,0 +1,49 @@
+CREATE TABLE "patients"(
+    "id" INT GENERATED ALWAYS AS IDENTITY,
+    "name" VARCHAR(255) NOT NULL,
+    "date_of_birth" DATE NOT NULL,
+    PRIMARY KEY(id)
+);
+
+CREATE TABLE "medical_histories"(
+    "id" INT GENERATED ALWAYS AS IDENTITY,
+    "admitted_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "patient_id" INTEGER NOT NULL,
+    "status" VARCHAR(255) NOT NULL,
+    PRIMARY KEY(id)
+);
+
+CREATE TABLE "invoices"(
+    "id" INT GENERATED ALWAYS AS IDENTITY,
+    "total_amount" DECIMAL(8, 2) NOT NULL,
+    "generated_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "payed_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "medical_history_id" INTEGER NOT NULL,
+    PRIMARY KEY(id)
+);
+
+CREATE TABLE "innvoice_items"(
+    "id" INT GENERATED ALWAYS AS IDENTITY,
+    "unit_price" DECIMAL(8, 2) NOT NULL,
+    "quantity" INTEGER NOT NULL,
+    "total_price" DECIMAL(8, 2) NOT NULL,
+    "invoice_id" INTEGER NOT NULL,
+    "treatment_id" INTEGER NOT NULL,
+    PRIMARY KEY(id)
+);
+
+CREATE TABLE "treatments"(
+    "id" INT GENERATED ALWAYS AS IDENTITY,
+    "type" VARCHAR(255) NOT NULL,
+    "name" VARCHAR(255) NOT NULL,
+    PRIMARY KEY(id)
+);
+
```

Comment:
- [ ] Kindly there's a many-to-many relationship between `medical_histories` and `treatments` should be created. You can create `medical_histories_has_treatments` table and add to it the Foreign keys.
- [ ] After that you need to add these two Foreign keys indexes, to speed the performance.

Reply:
Asked change implemented